library/core/src/ops/range.rs

```diff
@@ -6,6 +6,9 @@ use crate::hash::Hash;
 /// `RangeFull` is primarily used as a [slicing index], its shorthand is `..`.
 /// It cannot serve as an [`Iterator`] because it doesn't have a starting point.
 ///
+/// `RangeFull` is a [*zero-sized type*][ZST], which means it is a singleton –
+/// there is only one instance of the `RangeFull` type.
```

Comment:
suggestion: I would avoid the "singleton" and "instance" language here, since one *can* have `(.., 4, ..)` and such which could easily be argued to be two instances.  (Just because something is a ZST doesn't mean that their addresses are always `ptr::eq`, which is demonstrated most clearly by the slice iterators.)

Maybe just leave the mention off?  It's defined as `pub struct RangeFull;`, so I don't think we need to say this on every such type -- people can look it up in the book if they're curious.  (For example, `fmt::Error` doesn't say it's a ZST.)

Reply:
Makes sense. I think I added this because I say "This is **the** `RangeFull`" later on, and I thought this might make it clearer to people why I say "the": that there's only one possible value. But I can remove it :)